### ISSUE TITLE
EXE.xml 'System.IO.DirectoryNotFoundException'

### ISSUE THREAD

**Issue by KrazyKerbalnaut** [ResetXPDR/MSFS2020_AutoFPS#5]
As the title suggests, when trying to add "Auto-start with MSFS", I get: Error, Exception 'System.IO.DirectoryNotFoundException' during AutoStartExe. Configuring without this option it installs correctly.

**Comment by ResetXPDR** [comment]
Is the error message coming from MSFS or the app? If MSFS, can you please copy paste the section of your EXE.xml pertaining to this app, eg. 
    <Launch.Addon>

        <Disabled>False</Disabled>

        <ManualLoad>False</ManualLoad>

        <Name>MSFS2020_AutoFPS</Name>

        <Path>C:\Users\<your username>\AppData\Roaming\MSFS2020_AutoFPS\bin\MSFS2020_AutoFPS.exe</Path>

    </Launch.Addon>

**Comment by pandadoudou** [comment]
Hello, I have the same during the installation.
![image](https://github.com/ResetXPDR/MSFS2020_AutoFPS/assets/28102855/92366ced-c2d6-41af-bae7-80c946f2228f)

**Comment by KrazyKerbalnaut** [comment]
> Is the error message coming from MSFS or the app? If MSFS, can you please copy paste the section of your EXE.xml pertaining to this app, eg.
> 
> ```
> <Launch.Addon>
> 
>     <Disabled>False</Disabled>
> 
>     <ManualLoad>False</ManualLoad>
> 
>     <Name>MSFS2020_AutoFPS</Name>
> 
>     <Path>C:\Users\<your username>\AppData\Roaming\MSFS2020_AutoFPS\bin\MSFS2020_AutoFPS.exe</Path>
> 
> </Launch.Addon>
> ```

No, it isn't MSFS, this is during the installation executable for the app. Using the "Do not confugure Auto-Start", I can launch the app on my own and use it that way.

**Comment by ResetXPDR** [comment]
Thanks for clarifying and the screen shots. That error message indicates that the installer is unable to locate your EXE.xml file in the usual directories where MSFS is installed for Steam and MS Store versions. Do you by chance happen to have a custom install location?

In any case, you can still run the app manually without the autostart working, as @pandadoudou discovered. Until I can work out a fix, if you want the app to autostart you are going to have to locate EXE.xml on your system and edit it include what I wrote in my initial response to this issue.

**Comment by KrazyKerbalnaut** [comment]
Thats

> Thanks for clarifying and the screen shots. That error message indicates that the installer is unable to locate your EXE.xml file in the usual directories where MSFS is installed for Steam and MS Store versions. Do you by chance happen to have a custom install location?
> 
> In any case, you can still run the app manually without the autostart working, as @pandadoudou discovered. Until I can work out a fix, if you want the app to autostart you are going to have to locate EXE.xml on your system and edit it include what I wrote in my initial response to this issue.

No, everything was left as default. I did have to just barely reinstall MSFS but I’ve run it several times before ever running into this issue. When I did install it, I was surprised that the new default location is in “C:/XboxGames” or something similar. I’ve turned off my computer for the night but I can let you know for sure tomorrow. However, all the other folders are the same… or “AppData/Local/Packages/Microsoft. FlightSimulator_8wekyb3d8bbwe.”

**Comment by ResetXPDR** [comment]
@KrazyKerbalnaut The installer only uses default install locations at this time. When you do fire up your computer again could you please locate EXE.xml and tell me where you found it so I can look at the code and see if there is something I can do to avert such an issue in the future. Thanks.

**Comment by KrazyKerbalnaut** [comment]
> @KrazyKerbalnaut The installer only uses default install locations at this time. When you do fire up your computer again could you please locate EXE.xml and tell me where you found it so I can look at the code and see if there is something I can do to avert such an issue in the future. Thanks.

That sounds like a plan.

**Comment by KrazyKerbalnaut** [comment]
I am having a bout of insomnia so I went ahead and just searched my entire C:\ for the file as it isn't in the default location. Turns out the file doesn't exist. After some preliminary research, it appears this file doesn't exist until an addon requests to be started with the game? Maybe? Here is a discussion about a clean install that also did not have the exe.xml.

[https://forums.flightsimulator.com/t/missing-exe-xml-file/516723/30](https://forums.flightsimulator.com/t/missing-exe-xml-file/516723/30)
